### PR TITLE
#164798923 Users should be able to bookmark articles

### PR DESCRIPTION
**Pull request by samorahno**
#### Description
Creates the endpoint that gives a user the ability to bookmark articles
#### Type of change
- [x]  Non-breaking change(introduces new endpoint to codebase)
#### How Has This Been Tested?
- All tests are currently passing
- It was also tested on Postman

##### Screenshots
Screen Shot 2019-04-11 at 10 18 15 AM

#### Checklist:
- [x]  Create `bookmarks` model
- [x] Create association between `users` and `bookmarks` models
- [x] Create association between `articles` and `bookmarks` models
- [x] Write tests
- [x] Implement endpoint
#### PIVOTAL BOARD
[#164798923](https://www.pivotaltracker.com/story/show/164798923)

### CODE REVIEW

**Review by TheDrizzyWay — APPROVED** [review]
This looks pretty good. Well done

File: server/test/bookmark.spec.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      .post('/api/v1/auth/login')
+      .send({
+        email: 'mofe@okoro.com',
+        password: 'soldier'
+      });
+
+    userToken = userResponse.body.data.token;
+  });
+
+  describe('POST /api/v1/', () => {
```
Comment (chukwudidu) [inline server/test/bookmark.spec.js]: Good job. Please complete the `describe` route here for easy readability.

File: server/controllers/bookmark.contoller.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+   * @param {object} req
+   * @param {object} res
+   * @returns {object} response
+  */
+  static async createOrRemoveBookmark(req, res) {
+    const {
+      articleId
+    } = req.params;
+    const { id } = req.user;
+    const isArticleExists = await Article.findOne({ where: { id: articleId } });
```
Comment (chukwudidu) [inline server/controllers/bookmark.contoller.js:21]: Good one making this check.

**Review by chukwudidu — CHANGES_REQUESTED** [review]

File: server/controllers/bookmark.contoller.js (hunk 2 above)
Comment (davidokonji) [inline server/controllers/bookmark.contoller.js:21]: @samorahno There is a util function to check if the article does not exist, why not use that as a middleware?

File: server/migrations/20190411042815-create-bookmark.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        allowNull: false,
+        type: Sequelize.DATE,
+        defaultValue: Sequelize.fn('now')
+      }
+    });
+  },
+  down: (queryInterface, Sequelize) => {
+    return queryInterface.dropTable('Bookmarks');
+  }
+};
```
Comment (davidokonji) [inline server/migrations/20190411042815-create-bookmark.js]: @samorahno  EOF rules, add an end of a line

File: server/test/bookmark.spec.js (hunk 1 above)
Comment (davidokonji) [reply server/test/bookmark.spec.js]: @samorahno Yes like @cupidcoder said, please make the `describe()` more descriptive for better readability

**Review by geekdanitek — APPROVED** [review]

File: server/migrations/20190411042815-create-bookmark.js (hunk 3 above)
Comment (samorahno) [reply server/migrations/20190411042815-create-bookmark.js]: This will be fixed @davidokonji

File: server/test/bookmark.spec.js (hunk 1 above)
Comment (samorahno) [reply server/test/bookmark.spec.js]: Noted. I will complete the ```describe``` route

File: server/controllers/bookmark.contoller.js (hunk 2 above)
Comment (samorahno) [reply server/controllers/bookmark.contoller.js:21]: @davidokonji you are right, but the util function existing does not fit into this. it uses req.body

Comment (Chiazokam) [reply server/controllers/bookmark.contoller.js:21]: Yes, I observed same @davidokonji. The util function gets id from the body but he is getting from the params

**Review by Chiazokam — APPROVED** [review]

**Pull request merged**